src/java.desktop/share/classes/com/sun/imageio/plugins/tiff/TIFFImageReader.java

```diff
@@ -728,6 +728,11 @@ public Iterator<ImageTypeSpecifier> getImageTypes(int imageIndex) throws IIOExce
                             extraSamples,
                             colorMap);
 
+            if(itsRaw.getColorModel().hasAlpha() &&
+                    getCompression() == BaselineTIFFTagSet.COMPRESSION_JPEG) {
+                throw new IIOException("JPEG compressed tiles with alpha channel are not supported.");
+            }
```

Comment:
Well .. if we ever (find a way to) support this along with CMYK then someone will have to remember to
remove this condition.

We do not use "if(" we always use "if ("

Reply:
> Well .. if we ever (find a way to) support this along with CMYK then someone will have to remember to
remove this condition.

If someone adds the support to the JPEG classes, omitting TIFF, it might so happen that the TIFF is unnecessarily restricted. A basic code usage search reveals the `TIFFImageReader` class to me though, so I would have noticed if it was me refactoring the JPEG code. Not sure. Is it a problem?

> We do not use "if(" we always use "if ("
Fixed. Squashed commits.